File: Sources/ATTNPurchaseEvent.m
//
//  ATTNPurchaseEvent.m
//  attentive-ios-sdk
//
//  Created by Wyatt Davis on 12/7/22.
//

#import "ATTNEvent.h"
#import "ATTNPurchaseEvent.h"
#if __has_include(<attentive_ios_sdk_framework/ATTNSDK.h>)
#import "attentive_ios_sdk_framework/attentive_ios_sdk_framework-Swift.h"
#else
// Load the headers from the attentive-ios-sdk Pod
#import "attentive_ios_sdk/attentive_ios_sdk-Swift.h"
#endif

NS_ASSUME_NONNULL_BEGIN

@implementation ATTNPurchaseEvent

- (instancetype)initWithItems:(NSArray<ATTNItem *> *)items order:(ATTNOrder *)order {
  if (self = [super init]) {
    self->_items = items;
    self->_order = order;
    
    
    ATTCart * cart = [[ATTCart alloc] initWithCartId:@"test" andCartCoupon:@"test"];
    cart.cartCoupon = @"coupon";
  }

  return self;
}

@end

NS_ASSUME_NONNULL_END

File: Sources/Public/Events/ATTNPurchaseEvent.swift
//
//  ATTNPurchaseEvent.swift
//  attentive-ios-sdk-framework
//
//  Created by Vladimir - Work on 2024-05-30.
//

import Foundation

@objc(ATTNPurchaseEvent)
public class ATTNPurchaseEvent: NSObject, ATTNEvent {
  @objc public let items: [ATTNItem]
  @objc public let order: ATTNOrder
  @objc public var cart: ATTNCart?

  @objc(initWithItems:order:)
  public init(items: [ATTNItem], order: ATTNOrder) {
    self.items = items
    self.order = order
    super.init()
  }

  private override init() {
    fatalError("init() has not been implemented")
  }
}
